roiextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -429,133 +416,143 @@ def get_property_data(self, property_name):
         return ret_val
 
     @staticmethod
-    def write_segmentation(segext_obj, savepath, metadata_dict, **kwargs):
-        if isinstance(metadata_dict, str):
-            with open(metadata_dict, 'r') as f:
-                metadata_dict = yaml.safe_load(f)
-
+    def write_segmentation(segext_obj, savepath, **kwargs):
+        if not kwargs.get('metadata'):
+            raise Exception('provide metadata as argument')
+        metadata = kwargs.get('metadata')
+        if segext_obj.extractor_name == 'MultiSegmentationExtractor':
+            segext_objs = segext_obj.segmentations
+            metadata_list = metadata
+        else:
+            metadata_list = [metadata]
+            segext_objs = [segext_obj]
+        source_path = segext_objs[0].filepath
+        print(f'writing nwb for {segext_objs[0].extractor_name}\n')
+        if isinstance(metadata_list[0], str):
```

Comment:
the list should be in ImagingPlane, and that list will contain information about individual planes. The session-level information should only be defined once.